internal/ssh/ssh.go

```diff
@@ -19,10 +20,18 @@ import (
 // juju_ssh_default_port is the default port we expect the juju controllers to respond on.
 const juju_ssh_default_port = 17022
 
-// Resolver is the interface with the methods needed by the ssh jump server to route request.
-type Resolver interface {
+type publicKeySSHUserKey struct{}
+
+// SSHManager is the interface with the methods needed by the ssh jump server to route request.
+type SSHManager interface {
 	// AddrFromModelUUID is the method to resolve the address of the controller to contact given the model UUID.
-	AddrFromModelUUID(ctx context.Context, user openfga.User, modelUUID string) (string, error)
+	AddrFromModelUUID(ctx context.Context, user *openfga.User, modelTag names.ModelTag) (string, error)
+
+	// FetchIdentity
+	FetchIdentity(ctx context.Context, id string) (*openfga.User, error)
+
+	// VerifyPublicKey verifies the identityName is
+	VerifyPublicKey(ctx context.Context, user *openfga.User, fingerprint string) (bool, error)
```

Comment:
I think we mentioned that we could discuss this in a separate PR, but verifying the public key, arguably should be done by using the raw key rather than the fingerprint.